tedge_config/tests/test_tedge_config.rs

```diff
@@ -29,7 +29,7 @@ connect = "false"
     let config =
         TEdgeConfigRepository::new_with_defaults(config_location, config_defaults).load()?;
 
-    assert_eq!(config.query(DeviceIdSetting)?, "ABCD1234");
```

Comment:
Should we keep `id = "ABCD1234"` in the line 11? I guess not?

Reply:
We can remove it or not. The result must be the same in both cases.

* I added the test `test_any_device_id_provided_by_the_configuration_is_ignored()` to be sure that this fix doesn't break previous installation where the device id is actually stored in the configuration. One must be able to read the configuration file despite the extra line.
*  I added 3 others to check the behavior, when there is no extra device.id in the file (testing when there is a certificate, a path defined but no certificate, a path to something that is no a certificate).

Reply:
I saw `test_any_device_id_provided_by_the_configuration_is_ignored()`, that's why we don't need to test if the configuration is ignored in other test cases in my opinion.